data-prepper-core/src/integrationTest/java/org/opensearch/dataprepper/integration/HeadlessPipelinesIT.java

```diff
@@ -0,0 +1,133 @@
+import static org.hamcrest.Matchers.empty;
+
+class HeadlessPipelinesIT {
+    private static final Logger LOG = LoggerFactory.getLogger(ProcessorPipelineIT.class);
+    private static final String IN_MEMORY_IDENTIFIER_DLQ = "PipelineDLQIT";
+    private static final String IN_MEMORY_IDENTIFIER_FORWARD = "ForwardPipelineIT";
+    private static final String PIPELINE_DLQ_TEST_CONFIGURATION= "pipeline-dlq.yaml";
+    private static final String FORWARD_PIPELINE_TEST_CONFIGURATION= "forward-pipeline.yaml";
+    private DataPrepperTestRunner dataPrepperTestRunner;
+    private InMemorySourceAccessor inMemorySourceAccessor;
+    private InMemorySinkAccessor inMemorySinkAccessor;
+
+    private void createPipeline(final String pipelineConfiguration) {
+        dataPrepperTestRunner = DataPrepperTestRunner.builder()
+                .withPipelinesDirectoryOrFile(pipelineConfiguration)
+                .build();
+        inMemorySourceAccessor = dataPrepperTestRunner.getInMemorySourceAccessor();
+        inMemorySinkAccessor = dataPrepperTestRunner.getInMemorySinkAccessor();
+        dataPrepperTestRunner.start();
+        LOG.info("Started test runner.");
+    }
+
+    @AfterEach
+    void tearDown() {
+        LOG.info("Test tear down. Stop the test runner.");
+        dataPrepperTestRunner.stop();
+    }
+
+    @Test
+    void dlq_pipeline_test() {
+        createPipeline(PIPELINE_DLQ_TEST_CONFIGURATION);
+
+        final int recordsToCreate = 200;
+        final List<Record<Event>> inputRecords = IntStream.range(0, recordsToCreate)
+                .mapToObj(i -> UUID.randomUUID().toString())
+                .map(JacksonEvent::fromMessage)
+                .map(Record::new)
+                .collect(Collectors.toList());
+
+        LOG.info("Submitting a batch of record.");
+        inMemorySourceAccessor.submit(IN_MEMORY_IDENTIFIER_DLQ, inputRecords);
+
+        await().atMost(400, TimeUnit.MILLISECONDS)
+                .untilAsserted(() -> {
+            assertThat(inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER_DLQ), not(empty()));
+        });
+
+        assertThat(inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER_DLQ).size(), equalTo(recordsToCreate));
+
+        final List<Record<Event>> sinkRecords = inMemorySinkAccessor.get(IN_MEMORY_IDENTIFIER_DLQ);
+
+        for (int i = 0; i < sinkRecords.size(); i++) {
+            final Record<Event> inputRecord = inputRecords.get(i);
+            final Record<Event> sinkRecord = sinkRecords.get(i);
+            assertThat(sinkRecord, notNullValue());
+            final Event recordData = sinkRecord.getData();
+            assertThat(recordData, notNullValue());
+            assertThat(
+                    recordData.get("message", String.class),
+                    equalTo(inputRecord.getData().get("message", String.class)));
+            assertThat(recordData.get("test1", String.class),
+                    equalTo(null));
+
+        }
+    }
+
+    @Test
+    void pipeline_forward_test() {
```

Comment:
While both sink forwarding and DLQ use similar code underneath, they are different end functions. I think we should have different IT test suites for each to reflect that.

Reply:
I have one above for DLQ and this one is for forwarding. Is that what you meant?

Reply:
Oh. I can split them into two